addons/payment/models/payment_acquirer.py

```diff
@@ -336,6 +337,21 @@ def write(self, vals):
         self._check_required_if_provider()
         return result
 
+    def unlink(self):
+        """ Prevent the deletion of the payment acquirer if it has xmlid, because in that
+        case the record might be used in some modules.
+        """
+        external_ids = self.get_external_id()
+        for acquirer in self:
+            external_id = external_ids[acquirer.id]
```

Comment:
Nitpicking: this could be simplified with a `.items()`.
```suggestion
        for acquirer, xmlid in self.get_external_id().items():
```

Reply:
Hello @AntoineVDV 
If we apply this change we can get acquirer as id in int.
When we try to access `acquirer.name` in UserError message it generate error.
so can we use `self.browse(acquirer).name` instead of it?.

Thanks!

Reply:
Oh sorry, I didn't pay attention to that. Forget my suggestion then :)